synapse/handlers/oidc.py

```diff
@@ -117,7 +117,8 @@ async def load_metadata(self) -> None:
         for idp_id, p in self._providers.items():
             try:
                 await p.load_metadata()
-                await p.load_jwks()
+                if not p._uses_userinfo:
+                    await p.load_jwks()
```

Comment:
Problem is that we are raising if we call load_jwks and there is no jwks_uri in the OP metadata (which can happen)
Before, we were just returning an empty key set if the userinfo endpoint was used, which was fine because we never tried to verify ID tokens in that case.
Now, we verify ID tokens if they are present, even if we are using the userinfo endpoint, so we can't just return an empty set, but we still want to raise if we try to verify an ID token while there is no jwks_uri 
This here is just to crash if Synapse can't figure out the OP metadata & keyset on startup. We don't want it to crash if the OP does not have keys (like GitHub) because they are using the userinfo endpoint instead

Reply:
Thanks!